hstream/src/HStream/Server/Handler.hs

```diff
@@ -134,58 +135,69 @@ executeQueryHandler sc@ServerContext{..} (ServerNormalRequest _metadata CommandQ
-                (fromMaybe "127.0.0.1" $ fromCOptionString (lookup "host"     cOptions))
-                (fromMaybe "9000"      $ fromCOptionString (lookup "port"     cOptions))
-                (fromMaybe ""          $ fromCOptionString (lookup "password" cOptions))
-                False
-                (fromMaybe "default"   $ fromCOptionString (lookup "database" cOptions))
-              return $ Right $ clickHouseSinkConnector cli
-            "mysql" -> do
-              conn <- MySQL.connect $ MySQL.ConnectInfo
-                (fromMaybe "127.0.0.1" $ fromCOptionStringToString   (lookup "host" cOptions))
-                (fromMaybe 3306        $ fromCOptionIntToPortNumber  (lookup "port" cOptions))
-                (fromMaybe "mysql"     $ fromCOptionString       (lookup "database" cOptions))
-                (fromMaybe "root"      $ fromCOptionString       (lookup "username" cOptions))
-                (fromMaybe "password"  $ fromCOptionString       (lookup "password" cOptions))
-                33
-              return $ Right $ mysqlSinkConnector conn
-            _ -> return $ Left "unsupported sink connector type"
-        _ -> return $ Left "invalid type in connector options"
       MkSystemTime timestamp _ <- getSystemTime'
-      let cid = CB.pack $ showHex timestamp (T.unpack cName)
+      let cid = CB.pack $ T.unpack cName
           cinfo = P.Info (ZT.pack $ T.unpack $ TL.toStrict commandQueryStmtText) timestamp
-      case sk of
-        Left err -> returnErrRes err
-        Right connector -> case streamM of
-          Just (ConstantString stream) -> do
-            streamExists <- doesStreamExists scLDClient (transToStreamName $ T.pack stream)
-            case streamExists of
-              False -> returnErrRes $ "Stream " <> TL.pack stream <> " doesn't exist"
-              True -> do
-                ldreader <- newLDReader scLDClient 1000 Nothing
-                let sc = hstoreSourceConnectorWithoutCkp scLDClient ldreader
-                P.withMaybeZHandle zkHandle $ P.insertConnector cid cinfo
-                subscribeToStreamWithoutCkp sc (T.pack stream) Latest
-                _ <- async $ do
-                  P.withMaybeZHandle zkHandle $ P.setConnectorStatus cid P.Running
-                  forever $ do
-                    records <- readRecordsWithoutCkp sc
-                    forM_ records $ \SourceRecord {..} ->
-                      writeRecord connector $ SinkRecord (T.pack stream) srcKey srcValue srcTimestamp
-                returnOkRes
-          _ -> returnErrRes "stream name missed in connector options"
+      shouldCreate <- case ifNotExist of
+          True -> do
+              s <- try $ P.withMaybeZHandle zkHandle $ P.getConnectorStatus cid
+              case s of
+                Left FailedToGet -> return True -- Not Exists
+                Right _          -> return False -- Exists
+          False -> return True -- Don't care
```

Comment:
The meaning of `isNotExist` does not match its behavior.

Reply:
Would you like to make a more detailed explaination.